core/src/main/java/org/apache/carbondata/core/datastore/ReusableDataBuffer.java

```diff
@@ -47,7 +47,8 @@
    */
   public byte[] getDataBuffer(int requestedSize) {
     if (dataBuffer == null || requestedSize > size) {
-      this.size = requestedSize + ((requestedSize * 30) / 100);
+      this.size = requestedSize + ((requestedSize / 100) * 30);
```

Comment:
No. We can't change the requested size. If we change the requested_size like 
```requestedSize = requestedSize + ((requestedSize / 100) * 30) ``` 
we might get a size lesser than the original requested size because (requestedSize / 100) * 30 can go out of range for int which will return a negative value.
So we need to compare Original requested size and new size.

Reply:
i think (requestedSize / 100) * 30 cann't go out of range for int ,but requestedSize + ((requestedSize / 100) * 30) can
if you consider the this.size is negative, i suggest to add a comment to explain this

Reply:
Actually this is happening for one varchar scenario that's why need for this PR.  I'll add a comment to explain.

Reply:
What if the requestedSize is less than 100? requestedSize/100 will be ZERO, is this intended?

Reply:
If the requestedSize is less than 100 the buffer size won't get incremented by 30% and will be like,
``` this.size = requestedSize ```
which should not be a problem.

Reply:
Done.